onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -31,28 +31,28 @@ class SVMCommon {
 
   float kernel_dot(const T* A, int64_t a, const std::vector<float>& B, int64_t b, int64_t len, KERNEL k) const {
     float sum = 0.f;
+    float* pA = (float*)&A[a];
+    float* pB = (float*)&B[b];
     if (k == KERNEL::POLY) {
-      for (int64_t i = 0; i < len; i++) {
-        sum += B[b + i] * static_cast<float>(A[a + i]);
-      }
+      for (int64_t i = len; i > 0; --i, ++pA, ++pB)
+        sum += *pB * static_cast<float>(*pA);
```

Comment:
>static_cast<float> [](start = 21, length = 18)

Is this casting necessary?

Reply:
fixed

Reply:
How do you fix this? Is `double` still supported? Or the new op will only work with `float`?

---
In reply to: [269168097](https://github.com/Microsoft/onnxruntime/pull/101#discussion_r269168097) [](ancestors = 269168097)